From dacc86d41bf10b4aa7f3f5d7dc2096b2e6b6bc17 Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Fri, 18 Aug 2023 17:13:28 +0200
Subject: [PATCH 0576/1141] ARM: dts: stm32: populate all timer counter nodes
 on stm32mp13

Counter driver originally had support limited to quadrature interface and
simple counter. It has evolved, so add remaining counter nodes in stm32
timer nodes.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: Ia461010a7a980292ad4e0f74a27f4c78a5282be4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/323456
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm/boot/dts/stm32mp131.dtsi | 40 +++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

--- a/arch/arm/boot/dts/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/stm32mp131.dtsi
@@ -330,6 +330,11 @@
 			dma-names = "up";
 			status = "disabled";
 
+			counter {
+				compatible = "st,stm32-timer-counter";
+				status = "disabled";
+			};
+
 			timer@5 {
 				compatible = "st,stm32h7-timer-trigger";
 				reg = <5>;
@@ -350,6 +355,11 @@
 			dma-names = "up";
 			status = "disabled";
 
+			counter {
+				compatible = "st,stm32-timer-counter";
+				status = "disabled";
+			};
+
 			timer@6 {
 				compatible = "st,stm32h7-timer-trigger";
 				reg = <6>;
@@ -1268,6 +1278,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM12_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -1293,6 +1308,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM13_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -1318,6 +1338,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM14_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -1348,6 +1373,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM15_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -1376,6 +1406,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM16_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -1404,6 +1439,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM17_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
